DigitalLearningSolutions.Data/DataServices/CourseDataService.cs

```diff
@@ -127,5 +129,31 @@ public int GetNumberOfActiveCoursesAtCentreForCategory(int centreId, int adminCa
                 new { centreId, adminCategoryId }
             );
         }
+
+        public IEnumerable<CourseStatistics> GetCourseStatisticsAtCentreForCategoryId(int centreId, int categoryId)
+        {
+            return connection.Query<CourseStatistics>(
+                @$"SELECT
+                        cu.CustomisationID,
+                        cu.CentreID,
+                        cu.Active,
+                        cu.AllCentres,
+                        ap.ASPMenu,
+                        ap.ArchivedDate,
+                        ap.ApplicationName,
+                        cu.CustomisationName,
+                        {CourseHelper.DelegateCount},
+                        {CourseHelper.CompletedCount},
+                        {CourseHelper.AllAttempts},
+                        {CourseHelper.AttemptsPassed}
+                    FROM dbo.Customisations AS cu
+                    INNER JOIN dbo.CentreApplications AS ca ON ca.ApplicationID = cu.ApplicationID
+                    INNER JOIN dbo.Applications AS ap ON ap.ApplicationID = ca.ApplicationID
+                    WHERE (ap.CourseCategoryID = @categoryId OR @categoryId = 0) 
+                        AND (cu.CentreID = @centreId OR (cu.AllCentres = 1 AND ca.Active = 1))
+                        AND ca.CentreID = @centreId",

```

Comment:
Would it make sense to check this centre applications centre id condition within the `cu.AllCentres = 1 AND ca.Active = 1` statement (as I don't think it's relevant for other cases)?

Reply:
Unfortunately, it needs to be where it is (always present) because of the way the join works (we only join Customisations to CentreApplications on the ApplicationId). If it were moved into that other section, suddenly all the cases that match `cu.CentreID = @centreId` become joined to all the CentreApplication rows with the same ApplicationId. 

So for instance, for the CentreId 25, CategoryId 0, the query as is recovers 76 records. Moving that condition instead recovers 32,444 records. 

The reason we can't join the Customisations table to the CentreApplications table on CentreId as well, is that that would then exclude the generic Customisations with `AllCentres = 1` (where the Customisation might be for some other centre, but the CentreApplication still applies it to the centre in question)